code/mission/missiongoals.cpp

```diff
@@ -964,7 +964,7 @@ void mission_process_event( int event )
 			Mission_events[event].count = Directive_count;
 		}
 
-		if (Sexp_useful_number){
+		if (Assume_event_is_current){
 			Mission_events[event].flags |= MEF_CURRENT;
```

Comment:
Is the check here even needed?  I am guessing it was originally written this way because Assume_event_is_current could be marked as false in the intervening function calls.